### ISSUE TITLE
where to hook on server-side spreadsheet sandbox incoming commands?

### ISSUE THREAD

**Issue by chrismark** [audreyt/ethercalc#389]
Hello. 

So we have this spreadsheet and there's a special column that gets filled in with values from a database. Users are able to edit said special column and we'd want those edits to be reflected back to the database.

I'm wondering if you guys could give some pointers on where to have code that we could hook on to so we can inspect what cell is being edited with what value and react accordingly (updated to a database). Any suggestions welcome.

Thanks.

**Comment by audreyt** [comment]
`src/main.ls` line 439 has a precedent for the server side.

On the interface there is a "lock cell" button. Would it work if you'd lock the cells?

One can also hide the Unlock button with CSS (or by deleting `images/sc_unlock.png`).

**Comment by eddyparkinson** [comment]
Suggest you look at **emailonedit** formula

http://sheet.cellmaster.com.au/i3pqpufosc

  emailonedit should have most of what you need to do this. It, sends an email when a cell/range is edited.

  a formula to read/write a json would be a good addition.
  I know a few people who want such a formula.

**Comment by chrismark** [comment]
Thanks @audreyt. Will check that out. 
The special cells will be the only ones editable.

Thanks @eddyparkinson. That looks cool. Will try that out.

**Comment by eddyparkinson** [comment]
@chrismark you can add buttons with actions - for example,  =button("insert Row") =insert(trigger_cell,insert_range)  

It has triggers for edit,time and buttons, and several action formulas.

It also builds web apps:

 - webapp http://sheet.cellmaster.com.au/carbonoffset/app
 - spreadsheet http://sheet.cellmaster.com.au/carbonoffset

To clone: use the button in the top right of: http://sheet.cellmaster.com.au

**Comment by chrismark** [comment]
Hello @eddyparkinson. I was able to add the function I wanted (which rides on you emailonedit function) onto an installed node_module from a branch of your repo that seems to be not there anymore. It worked as I wanted. Thanks for tip and code comments.

I want to move the updates I made to a git repo. I forked your repo and did it on the website branch. The issue I'm having now is when I click the Blank Spreadsheet link, it redirects to a url with ?auth=something and just loops on that url. Do you happen to know why it does that? Thanks in advance for any suggestion.

**Comment by eddyparkinson** [comment]
Website branch
---
go here
https://github.com/chrismark/cellmaster/
 Change "Branch:master" to "Branch:Website"

You will get: https://github.com/chrismark/cellmaster/tree/website

My version: https://github.com/eddyparkinson/cellmaster/tree/website

Push changes
---
can you push your changes to https://github.com/chrismark/cellmaster/tree/website

You will then get a pull request button, and when you are ready send a pull request to me.

Blank Spreadsheet Loop
---
 the redirect is here https://github.com/eddyparkinson/cellmaster/blob/website/src/main.ls#L238-L244
 but I don't get a loop on my version.

**Comment by chrismark** [comment]
hi @eddyparkinson, thanks for replying.

ah the looping isn't on your branch but in my fork/branch *oneditdo* that I forgot to mention above, based on website branch -- https://github.com/chrismark/cellmaster/tree/oneditdo

**Comment by chrismark** [comment]
hi @eddyparkinson.

`cellmaster/oneditdo` seems to be working now. Don't know why previously doing `npm i` then `make` didn't work and had that looping issue. Now my issue is I get #VALUE! in cell where i put my formula. I'll just recheck code again and make sure all are there.

**Comment by eddyparkinson** [comment]
Did you work this out? 
Can I merge it in?

Sorry only just seen your message, I have had a busy 2 weeks, but things are back to normal now.

I looked at the code, looked good to me. Could not see an obvious reason for the #VALUE

**Comment by chrismark** [comment]
Hi @eddyparkinson. Yes I've worked it out and turns out I mistyped some codes. Thanks.

I've also updated the base code to have latest commits form ethercalc#master. Works great but we experienced an issue where right after filling spreadsheet (using curl, twice) then refreshing browser the spreadsheet goes bare. We also tested it on ones without my changes and same thing. Issue is here: https://github.com/audreyt/ethercalc/issues/425
